src/printer.h

```diff
@@ -27,6 +27,34 @@ class Precedence : public BaseVisitor<Precedence> {
         precedence = PrecedenceEnum::Pow;
     }
 
+    void bvisit(const Polynomial &x) {
+        if (x.dict_.size() == 1) {
+            auto it = x.dict_.begin();
+            if (it->second == 0) {
+                precedence = PrecedenceEnum::Atom;
+            }
+            else if (it->second == 1) {
+                if (it->first == 0 || it->first == 1) {
+                    precedence = PrecedenceEnum::Atom;
+                }
+                else {
+                    precedence = PrecedenceEnum::Pow;
+                }
+            }
+            else {
+                if (it->first == 0) {
+                    precedence = PrecedenceEnum::Atom;
```

Comment:
`it->first` is `uint`, we are handling only `uint` degrees now.
Are you trying to say that `-1` doesn't come under `Atom`, I'm not getting you I think.